pyked/converters.py

```diff
@@ -0,0 +1,741 @@
+                    spec['InChI'] = child.find('speciesLink').attrib['InChI']
+                except KeyError:
+                    # TODO: add InChI validator/search
+                    print('Warning: missing InChI for species ' + spec['species-name'])
+                    pass
+
+                # amount of that species
+                spec['amount'] = [float(child.find('amount').text)]
+
+                properties['composition']['species'].append(spec)
+
+                # check consistency of composition type
+                if not composition_type:
+                    composition_type = child.find('amount').attrib['units']
+                elif composition_type != child.find('amount').attrib['units']:
+                    raise KeyError('inconsistent initial composition units')
+            assert composition_type in ['mole fraction', 'mass fraction']
+            properties['composition']['kind'] = composition_type
+
+        elif name in ['temperature', 'pressure', 'pressure rise', 'compression time']:
+            field = name.replace(' ', '-')
+            units = elem.attrib['units']
+            if units == 'Torr':
+                units = 'torr'
+            quantity = 1.0 * unit_registry(units)
+            try:
+                quantity.to(property_units[field])
+            except pint.DimensionalityError:
+                raise KeyError('Error: units incompatible for property ' + name)
+
+            properties[field] = [' '.join([elem.find('value').text, units])]
+
+        else:
+            raise KeyError('Property ' + name + ' not supported as common property.')
+
+    return properties
+
+
+def get_ignition_type(root):
+    """Gets ignition type and target.
+
+    Args:
+        root (`etree.Element`): Root of ReSpecTh XML file
+
+    Returns:
+        properties (`dict`): Dictionary with ignition type/target information
+    """
+    ignition = {}
+    elem = root.find('ignitionType')
+
+    if elem is None:
+        raise AttributeError('missing ignitionType')
+
+    try:
+        ign_target = elem.attrib['target'].rstrip(';').upper()
+    except KeyError:
+        raise AttributeError('missing ignitionType/target')
+    try:
+        ign_type = elem.attrib['type']
+    except KeyError:
+        raise AttributeError('missing ignitionType/type')
+
+    # ReSpecTh allows multiple ignition targets
+    if len(ign_target.split(';')) > 1:
+        raise NotImplementedError('Multiple ignition targets not supported.')
+
+    # Acceptable ignition targets include pressure, temperature, and species
+    # concentrations
+    if ign_target == 'OHEX':
+        ign_target = 'OH*'
+    elif ign_target == 'CHEX':
+        ign_target = 'CH*'
+
+    if ign_target not in ['P', 'T', 'OH', 'OH*', 'CH*', 'CH']:
+        raise KeyError(ign_target + ' not valid ignition target')
+
+    if ign_type not in ['max', 'd/dt max', '1/2 max', 'min']:
+        raise KeyError(ign_type + ' not valid ignition type')
+
+    if ign_target == 'P':
```

Comment:
There are only two conversions, for pressure and temperature, and they are already in an `if-elif` block:
```
    if ign_target == 'P':
        ign_target = 'pressure'
    elif ign_target == 'T':
        ign_target = 'temperature'
```
unless I'm missing what you mean?